src/components/blockMenu/permissions.tsx

```diff
@@ -1,81 +1,312 @@
+import React, { useEffect, useState } from 'react'
+import { View, StyleSheet, SectionList } from 'react-native'
+import {
+	ActivityIndicator,
+	Appbar,
+	Avatar,
+	Button,
+	Divider,
+	List,
+	Snackbar,
+	Text,
+	Title,
+	TouchableRipple,
+} from 'react-native-paper'
+import { useMutation, useQuery } from 'urql'
+import { GET_PERMISSIONS, SET_PERMISSIONS, UPDATE_VISIBILITY } from '../../api/gql'
+import { Permission, User } from '../../api/types'
+import routes from '../../navigation/routes'
 import colors from '../../utils/colors'
+import { globalStyles } from '../../utils/styles'
+import { getInitials, textToColor } from '../../utils/utils'
+import { useActionSheet } from '@expo/react-native-action-sheet'
 
-export const Permissions = ({ route }: { route: any }) => {
+export const Permissions = ({ route, navigation }: { route: any; navigation: any }) => {
 	const menu: MenuComponent = route.params?.menu
 	let isPublic = menu?.permissions.public
 
+	const [full, setFull] = useState([])
+	const [edit, setEdit] = useState([])
+	const [view, setView] = useState([])
+	const [isLoading, setLoading] = useState(false)
+	const [snackbarVisible, setSnackbarVisible] = useState(false)
+	const { showActionSheetWithOptions } = useActionSheet()
+
 	type VisibilityResult = { updateVisibility: { id: number; public: boolean } }
 	type VisibilityRequest = { blockId: number; public: boolean }
 	const [visibilityResult, setVisibility] = useMutation<VisibilityResult, VisibilityRequest>(UPDATE_VISIBILITY)
 
+	type PermissionResult = { blockById: Permission }
+	type PermissionRequest = { id: number }
+	const [permissionResponse, getPermissions] = useQuery<PermissionResult, PermissionRequest>({
+		query: GET_PERMISSIONS,
+		variables: { id: menu.block_id },
+	})
+
+	type SetPermissionRequest = { blockId: number; full: Array<number>; edit: Array<number>; view: Array<number> }
+	const [, setPermissions] = useMutation<PermissionResult, SetPermissionRequest>(SET_PERMISSIONS)
+
+	useEffect(() => {
+		getPermissions()
+	}, [])
+
+	const permissionsData = [
+		{ title: 'Full', data: full },
+		{ title: 'Edit', data: edit },
+		{ title: 'View', data: view },
+	]
+
 	if (visibilityResult.data?.updateVisibility) {
 		isPublic = visibilityResult.data?.updateVisibility.public
 	}
 
+	useEffect(() => {
+		if (permissionResponse.data?.blockById) {
+			const permissions = permissionResponse.data?.blockById
+			setFull(permissions.full)
+			setEdit(permissions.edit)
+			setView(permissions.view)
+		}
+	}, [permissionResponse])
+
+	// Capturing data from the child component
+	React.useEffect(() => {
+		if (route.params?.user) {
+			const exists = view.some(e => e.id === route.params?.user.id)
+			if (!exists) {
+				setView(oldView => [...oldView, route.params?.user])
+			}
+		}
+	}, [route.params?.user])
+
+	React.useLayoutEffect(() => {
+		navigation.setOptions({
+			headerRight: () => {
+				return (
+					<Appbar.Action
+						icon='plus'
```

Comment:
I think it would work best if the button to add users was at the level where they are, since we also have visibility on this screen. Maybe next to the "User Permissions" heading?